public/index.html

```diff
@@ -14,26 +14,53 @@
 	<body>
 		<canvas id="canvas"></canvas>
 		<div class="controlpanel">
-			<button onclick="save(true)">Save</button>
-			<button onclick="generateSolidWall()">Generate Walls</button>
-			<button onclick="fixAll()">Fix All Rooms</button>
-			<input type="range" name="wallChance" value="25" min="0" max="100" step="10" oninput="this.title=this.value + '%'">
-			<label>
-				<input type="checkbox" id="showWalls">
-				Show Walls
-			</label>
-			<button onclick="autoGen(2,2)">Auto Gen (2x2)</button>
-			<button onclick="autoGen(4,2)">Auto Gen (4x2)</button>
-			<button onclick="autoGen(4,4)">Auto Gen (4x4)</button>
-			<label>
-				Mouse Tool
+			<div>
+				<div>Mouse Tool :</div>				
 				<select oninput="currentTool = this.value">
 					<option value="gen">Generate Rooms</option>
 					<option value="edit">Edit Room Terrain</option>
 					<option value="access">Open/Close Rooms</option>
 					<option selected value="block">No Tool Selected</option>					
 				</select>
+			<div>
+			<div>
+				<button onclick="generateSolidWall()">Generate Walls</button>
+			</div>
+			<div>
+				<button onclick="fixAll()">Fix All Rooms</button>
+			</div>
+			<div>
+				<div>Walls Chance</div>
+				<input type="range" name="wallChance" value="25" min="0" max="100" step="1" oninput="this.title=this.value + '%'">
+				
+			</div>
+			<div>
+				<div>2 Source chance</div>
+				<input type="range" name="twoSourcesChance" value="25" min="0" max="101" step="1" oninput="this.title=this.value + '%'">
```

Comment:
Last I heard, 101 is an odd max for a percentage ;)

Reply:
fixed to be proper 💯